.github/workflows/daily-test.yml

```diff
@@ -0,0 +1,55 @@
+name: Daily Test
+
+on:
+  schedule:
+    # More information on cron https://crontab.guru/
+    # GitHub actions is using UTC time. Scheduling action at 5 am PST
+    - cron: '0 13 * * *'
+
+  workflow_dispatch:
+
+env:
+  SAUCE_USERNAME: ${{ secrets.SAUCE_USERNAME }}
+  SAUCE_ACCESS_KEY: ${{ secrets.SAUCE_ACCESS_KEY }}
+  SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+
+jobs:
+  daily-test:
+    name: Amazon Chime iOS SDK Daily Test
+    runs-on: ubuntu-latest
+    outputs:
+      job-status: ${{ job.status }}
+
+    steps:
+      - name: Checkout test package
+        uses: actions/checkout@v4
+        with:
+          repository: awslabs/amazon-chime-sdk-apple
+          token: ${{ secrets.GH_INTEG_READ_ONLY_PAT }}
+          ref: automated-test-development
+
+      - name: Get latest prod demo app
+        run: |
+          wget -O amazon-chime-sdk-app.ipa ${{ secrets.DEMO_APP_DOWNLOAD_LINK }}
+      - name: Setup Node.js - 15.x
+        uses: actions/setup-node@v4
+        with:
+          node-version: 15.x
+
+      - name: Run tests
+        id: tests
+        run: |
+          id=$(curl -F 'payload=@amazon-chime-sdk-app.ipa' -F name=amazon-chime-sdk-app.ipa -u "${{ secrets.SAUCE_USERNAME }}:${{ secrets.SAUCE_ACCESS_KEY }}" 'https://api.us-west-1.saucelabs.com/v1/storage/upload' |jq '.item.id')
+          npm install
+          npm run build
+          npm run cma -- --app-url "sample.app" --log-level error --tag "@common" --app-id "${id}"
+      - name: Send Notification
+        uses: slackapi/slack-github-action@v1.25.0
+        if: always()
```

Comment:
do we need this if?

Reply:
yes, otherwise the step will be skipped if any above step fails